design-docs/001-virtual-chunk-improvements.md

```diff
@@ -0,0 +1,163 @@
+# Improvements to virtual chunks handling
+
+## What we need
```

Comment:
I understand that this design document pertains mostly to the `VirtualChunkContainer`, so I'm not sure if these are in scope but there are some interactions at the store level that may be worth mentioning. For example,

- Does not provide a barrier to distributed writes of virtual chunks (xref https://github.com/zarr-developers/VirtualiZarr/issues/308)
- Does not provide a barrier to combining virtual chunks of variable lengths in a single store (xref https://github.com/zarr-developers/VirtualiZarr/issues/12)
- Does not provide a barrier to combining virtual chunks with "inlined" data in a single store (xref https://github.com/zarr-developers/VirtualiZarr/issues/62)